kubectl rolling-update nil pointer dereference

@kubernetes/goog-ux I can provide the exact rc if you need a better repro. 

```
$ kubectl rolling-update glbc --container l7-lb-controller --image=gcr.io/google_containers/glbc:0.5.6 

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x760287]

goroutine 1 [running]:
k8s.io/kubernetes/pkg/runtime.Encode(0x0, 0x0, 0x7f992630d470, 0xc820020f00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/runtime/codec.go:45 +0xf7
k8s.io/kubernetes/pkg/api.HashObject(0x7f992630d470, 0xc820020f00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/api/helpers.go:157 +0xb2
k8s.io/kubernetes/pkg/kubectl.CreateNewControllerFromCurrentController(0x7f992630f410, 0xc82029d5a0, 0x0, 0x0, 0x15f9898, 0x7, 0x7ffd7dfadce4, 0x4, 0x0, 0x0, ...)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/rolling_updater.go:576 +0x754
k8s.io/kubernetes/pkg/kubectl/cmd.RunRollingUpdate(0xc82028c200, 0x7f9926348198, 0xc8200ba008, 0xc8202b0400, 0xc8202ac640, 0x1, 0x4, 0xc82029d8c0, 0x0, 0x0)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/rollingupdate.go:253 +0x422c
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdRollingUpdate.func1(0xc8202b0400, 0xc8202ac640, 0x1, 0x4)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/rollingupdate.go:87 +0x7d
github.com/spf13/cobra.(*Command).execute(0xc8202b0400, 0xc8202ac600, 0x4, 0x4, 0x0, 0x0)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:572 +0x869
github.com/spf13/cobra.(*Command).ExecuteC(0xc8202a0000, 0xc8202b0400, 0x0, 0x0)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:662 +0x54e
github.com/spf13/cobra.(*Command).Execute(0xc8202a0000, 0x0, 0x0)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:618 +0x2d
k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:28 +0xf0
main.main()
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:28 +0x28

goroutine 19 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x1df1460)
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x67
created by github.com/golang/glog.init.1
	/usr/local/google/home/beeps/goproj/src/k8s.io/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x297

goroutine 39 [syscall]:
os/signal.loop()
	/usr/lib/google-golang/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/lib/google-golang/src/os/signal/signal_unix.go:28 +0x37

goroutine 15 [IO wait]:
net.runtime_pollWait(0x7f9926315af0, 0x72, 0xc8200a4080)
	/usr/lib/google-golang/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8204966f0, 0x72, 0x0, 0x0)
	/usr/lib/google-golang/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8204966f0, 0x0, 0x0)
	/usr/lib/google-golang/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820496690, 0xc82043f000, 0x800, 0x800, 0x0, 0x7f9926344028, 0xc8200a4080)
	/usr/lib/google-golang/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200d0de0, 0xc82043f000, 0x800, 0x800, 0x0, 0x0, 0x0)
	/usr/lib/google-golang/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc82049da10, 0x7f9926315c00, 0xc8200d0de0, 0x5, 0x0, 0x0)
	/usr/lib/google-golang/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc820398580, 0x18bc717, 0x0, 0x0)
	/usr/lib/google-golang/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc820398580, 0xc8204d0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/google-golang/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7f9926384370, 0xc820398580, 0xc82047aaa8, 0xc8204d0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/lib/google-golang/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc82029c4a0, 0xc8204d0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82022c300)
	/usr/lib/google-golang/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82022c300, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/lib/google-golang/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc82047aa50)
	/usr/lib/google-golang/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/lib/google-golang/src/net/http/transport.go:685 +0xc78

goroutine 16 [select]:
net/http.(*persistConn).writeLoop(0xc82047aa50)
	/usr/lib/google-golang/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/lib/google-golang/src/net/http/transport.go:686 +0xc9d

```

--- a/pkg/kubectl/cmd/rollingupdate.go
+++ b/pkg/kubectl/cmd/rollingupdate.go
@@ -30,7 +30,6 @@ import (
 	"k8s.io/kubernetes/pkg/api/errors"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/v1"
-	"k8s.io/kubernetes/pkg/apimachinery/registered"
 	"k8s.io/kubernetes/pkg/kubectl"
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/resource"
@@ -235,7 +234,7 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 	// than the old rc. This selector is the hash of the rc, which will differ because the new rc has a
 	// different image.
 	if len(image) != 0 {
-		codec := registered.GroupOrDie(client.APIVersion().Group).Codec
+		codec := api.Codecs.LegacyCodec(client.APIVersion())
 		keepOldName = len(args) == 1
 		newName := findNewName(args, oldRc)
 		if newRc, err = kubectl.LoadExistingNextReplicationController(client, cmdNamespace, newName); err != nil {

